### PR TITLE
[feature] Indexer status

### PR DESCRIPTION
**Pull request by alecsg77**
Draft solution for #3292

The status will be

- `status:healthy`: the last request/test completed successfully. Status expires in 10 minutes.
- `status:failing`: last request/test failed. The status expires in min(1second * 2^retry, 24h)
- `status:unknown`: the status expired

All the reqest than generate an error are counting as failed request (network, parsing, login)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -585,25 +585,27 @@ To get all Jackett indexers including their capabilities you can use `t=indexers
...
 Filter | Condition
 -|-
-<code>type:<i><b>type</b></i></code> | where the indexer type is equal to <i><b>type</b></i>
-<code>tag:<i><b>tag</b></i></code> | where the indexer tags contains <i><b>tag</b></i>
-<code>lang:<i><b>tag</b></i></code> | where the indexer language start with <i><b>lang</b></i>
+`type:<type>` | where the indexer type is equal to `<type>`
+`tag:<tag>` | where the indexer tags contains `<tag>`
+`lang:<tag>` | where the indexer language start with `<lang>`
+`test:{passed|failed}` | where the last indexer test performed `passed` or `failed`
+`state:{healty|failing|unknown}` | where the indexer state is `healty` (succesfully operates in the last minutes), `failing` (generates errors in the recent call) or `unknown` (unused for a while)
```
Comment (ilike2burnthing) [inline README.md:598]:
```suggestion
`status:{healthy|failing|unknown}` | where the indexer state is `healthy` (successfully operates in the last minutes), `failing` (generates errors in the recent call) or `unknown` (unused for a while)
```

**Comment by ilike2burnthing** [comment]
@binhex [0.18.61](https://dev.azure.com/Jackett/0f0cb3d0-9b6b-4a86-876d-0c2e17c18801/_apis/build/builds/3495/artifacts?artifactName=drop&api-version=6.0&%24format=zip)

**Comment by binhex** [comment]
@ilike2burnthing @alecsg77 the 'Filter' works well for me in 0.18.61 linked above, i have tried all dropdown options and they filter correctly.

my only comment is perhaps people may want to combine the filters, for example 'type public + test passed', so perhaps checkboxes would be an idea instead of a drodown list, just a thought thats all, im fine with it as it is.

**Comment by ilike2burnthing** [comment]
Hi @alecsg77, is there anything else you're wanting to add to this PR?

**Comment by alecsg77** [comment]
Hi @ilike2burnthing. Sorry, I was too busy in the last period. I hope I can check it this week.

**Comment by ilike2burnthing** [comment]
No rush, just wanted to make sure you were waiting on it being merged as is.

**Comment by alecsg77** [comment]
The current implementation is exposed only on API side.
There is no way to use it on UI side.
The timing is not configurable.

If those limitations are acceptable, we can merge this PR and use other PRs to improve it.

**Comment by ilike2burnthing** [comment]
I'm ok with just having `test:` accessible in the UI as a filter, and leaving `status:` API-only, given it will change without the user knowing, and if they're setting something in the UI they are relying on it doing what they see (e.g. if it takes them 30s to enter their search term and choose categories, those indexers deemed `healthy` may be different than when they first opened the search).

Re: timing, is there away to tie the healthy status to twice the user set cache TTL? Currently it is twice the default cache TTL, but if a user changed that to say 75min, the cache would now outlive the healthy status. This would also give the user a way to control the healthy status (in a round about fashion).

The cache can be disabled, so I'm not sure if the TTL internally then becomes 0 (it doesn't in the UI at least), or it is just ignored.

Once you make changes you're happy with (or if they're unfeasible) mark as ready for review, I'll give it a last test and merge it.

**Comment by alecsg77** [comment]
I finally found the time to complete the changes.
When the cache is disabled, the cache TTL remains the same value, it's just a boolean flag.
I used the twice TTL configured value for the Healthy Status Validity, no matter the cache is enabled or not.

**Comment by alecsg77** [comment]
I finally found the time to complete the changes.
When the cache is disabled, the cache TTL remains the same value, it's just a boolean flag.
I used the twice TTL configured value for the Healthy Status Validity, no matter the cache is enabled or not.

**Comment by ilike2burnthing** [comment]
Thanks again!

**Pull request merged**